stellarphot/core.py

```diff
@@ -62,3 +70,305 @@ def __init__(self, gain=1.0 * u.electron / u.adu,
         self.gain = gain
         self.read_noise = read_noise
         self.dark_current = dark_current
+
+
+@autocheck_required_columns
+class BaseEnhancedTable(BaseTimeSeries):
+    """
+    A class to validate an `astropy.table.Table` table of astronomical data during
+    creation.
+
+    This is based on the `astropy.timeseries.BaseTimeSeries` class to
+    allow inheriting its ability to have required columns (but not requiring
+    the time column).  We extend this to allow for checking of the units for
+    the columns to match the table_description.
+
+    Parameters
+    ----------
+
+    table_description: dict
+        This is a dictionary where each key is a required table column name
+        and the value corresponding to each key is the required dtype
+        (can be None).  This is used to check the format of the input data
+        table.
+
+    data: `astropy.table.Table`
+        A table containing astronomical data of interest.  This table
+        will be checked to make sure all columns listed in table_description
+        exist and have the right runits. However, additional columns that
+        may be present in data but are not listed in table_description will
+        NOT be removed.
+
+    Attributes
+    ----------
+    data: `astropy.table.Table`
+        A table formatted to match the table_description formatting information
+        but containing no data.
+    """
+
+    def __init__(self, table_description, data):
+        # Handle parameters
+        self._table_description = table_description
+        self.data = data.copy()
+
+        # Build the data table
+        if not isinstance(self.data, Table):
+            raise TypeError(f"You must provide an astropy Table as data (it is type {type(self.data)}).")
+        else:
+            # Using the BaseTimeSeries class to handle the data table means required columns
+            # are checked.
+            super().__init__(data=self.data)
+
+            # Check that self.data exists now
+            if self.data is None:
+                raise ValueError(f"self.data should exist at this point, but doesn't, crap.")
+
+            # Confirm a proper table description is passed
+            if not isinstance(self._table_description, dict):
```

Comment:
I don't think we need to be this restrictive. The only thing we really need is that `table_description` is accessible via keys. We may want to make sure this is a dictionary for us, so we could do something like this up where you currently set `self._table_description = table_description` (lines below are untested, meant to illustrate the idea)

```python
try:
    self._table_description = {k: v for k, v in table_description.items()}
except AttributeError:
    raise TypeError(...)
```